program/src/processor.rs

```diff
@@ -1108,3 +1108,51 @@ pub fn process(program_id: &Pubkey, accounts: &[AccountInfo], input: &[u8]) -> P
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use {super::*, proptest::prelude::*};
+
+    proptest! {
+        #[test]
+        fn test_calculate_proposal_vote_threshold(
+            stake in 0u64..,
+            total_stake in 0u64..,
```

Comment:
This would probably be best served by a `prop_compose!` for a total and intermediate amount, ie:

```
    prop_compose! {
        pub fn total_and_intermediate(max_value: u64)(total in 0..=max_value)
                        (intermediate in 0..=total, total in Just(total))
                        -> (u128, u128) {
           (total, intermediate)
       }
    }
```

And then we can unwrap everything!

Reply:
Great, thanks - added! Was there a particular reason to return a tuple of `u128` instead of `u64`? It seemed to be a little simpler to return `u64`.

Reply:
Sorry, that was just bad copy paste, the change looks great!